lib/charms/operator_libs_linux/v2/snap.py

```diff
@@ -310,23 +310,32 @@ def _snap_daemons(
         except CalledProcessError as e:
             raise SnapError("Could not {} for snap [{}]: {}".format(args, self._name, e.stderr))
 
-    def get(self, key) -> str:
+    def get(self, key: str, use_json: bool = False) -> Any:
         """Fetch a snap configuration value.
 
         Args:
             key: the key to retrieve
+            use_json: (optional) retrieve values in JSON format. Default `False`
         """
-        return self._snap("get", [key]).strip()
+        if not use_json:
+            return self._snap("get", [key]).strip()
 
-    def set(self, config: Dict) -> str:
+        configs = json.loads(self._snap("get", ["-d", key]))
+        return configs.get(key)
+
+    def set(self, configs: Dict[str, Any], use_json: bool = False) -> str:
```

Comment:
We can't actually change the name of the `config` parameter without breaking backwards compatibility, as some people might be calling it by keyword (it's not a positional-only argument). And similar to above, I think we should change `use_json` to `typed`. So let's change the signature to:

```python
def set(self, config: Dict[str, Any], *, typed=False) -> str:
    ...
```

Reply:
Indeed, I didn't consider that people might calling it by keywords. Thanks for the suggestion